### PR TITLE
Adds SEPA Debit PaymentMethod

### PR DESCRIPTION
**Pull request by csabol-stripe**
## Summary

Adds bindings for SEPA Debit on PaymentMethod and adds examples for SEPA Debit and iDEAL.

## Motivation

New PaymentMethods!

## Testing

Tested w/ "Custom Integration" sample app

### CODE REVIEW

File: Example/Custom Integration/PaymentExampleViewController.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#import <UIKit/UIKit.h>
+
+#import <Stripe/Stripe.h>
+
+#import "BrowseExamplesViewController.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface PaymentExampleViewController : UIViewController <STPAuthenticationContext>
```
Comment (csabol-stripe) [inline Example/Custom Integration/PaymentExampleViewController.h]: so we can stop just copy-pasting as we add view controllers. Should definitely modernize this code more though

File: Stripe/Payments/STPPaymentHandler.m [hunk 2]
```diff
@@ -105,6 +110,7 @@ - (void)confirmPayment:(STPPaymentIntentParams *)paymentParams
         params.useStripeSDK = @YES;
     }
     [self.apiClient confirmPaymentIntentWithParams:params
+                                            expand:@[@"payment_method"]
```
Comment (csabol-stripe) [inline Stripe/Payments/STPPaymentHandler.m]: filed a followup for a general solution for this

File: Stripe/STPPaymentIntentParams.m [hunk 3]
```diff
@@ -12,6 +12,12 @@
 #import "STPPaymentMethodParams.h"
 #import "STPPaymentResult.h"
 
+@interface STPPaymentIntentParams ()
+
+@property (nonatomic, nullable, readonly) NSDictionary *mandateData;
```
Comment (csabol-stripe) [inline Stripe/STPPaymentIntentParams.m]: followup to expose this publically

File: Example/Custom Integration/SEPADebitExampleViewController.m [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+//
+//  Created by Cameron Sabol on 10/7/19.
+//  Copyright © 2019 Stripe. All rights reserved.
+//
+
+#import "SEPADebitExampleViewController.h"
+
+#import "MyAPIClient.h"
+
+@interface SEPADebitExampleViewController ()
```
Comment (yuki-stripe) [inline Example/Custom Integration/SEPADebitExampleViewController.m]: Intro blurb like some of the other examples + a link to the doc would be useful I think

File: Example/Custom Integration/SEPADebitExampleViewController.m [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    self.title = @"SEPA Debit";
+
+    [self.payButton setTitle:@"Pay with SEPA Debit" forState:UIControlStateNormal];
+    [self.payButton sizeToFit];
+
+    UILabel *mandateAuthLabel = [[UILabel alloc] init];
+    mandateAuthLabel.numberOfLines = 0;
+    mandateAuthLabel.font = [UIFont preferredFontForTextStyle:UIFontTextStyleCallout];
+    mandateAuthLabel.textAlignment = NSTextAlignmentCenter;
+    mandateAuthLabel.text = @"By providing your IBAN and confirming this payment, you are authorizing EXAMPLE COMPANY NAME and Stripe, our payment service provider, to send instructions to your bank to debit your account and your bank to debit your account in accordance with those instructions. You are entitled to a refund from your bank under the terms and conditions of your agreement with your bank. A refund must be claimed within 8 weeks starting from the date on which your account was debited.";
```
Comment (yuki-stripe) [inline Example/Custom Integration/SEPADebitExampleViewController.m]: as a user reading this, I'd be wondering what the requirements for this text are.  Not sure if the docs will cover that or if we should add a comment here

File: Stripe/Payments/STPPaymentHandler.m [hunk 6]
```diff
@@ -271,6 +279,19 @@ - (void)handleNextActionForSetupIntent:(NSString *)setupIntentClientSecret
 
 #pragma mark - Private Helpers
 
++ (BOOL)_isProcessingIntentSuccessForType:(STPPaymentMethodType)type {
```
Comment (yuki-stripe) [inline Stripe/Payments/STPPaymentHandler.m]: Can we add a comment

File: Stripe/STPAPIClient+Private.h [hunk 7]
```diff
@@ -129,6 +129,18 @@ fromCustomerUsingKey:(STPEphemeralKey *)ephemeralKey
 
 @end
 
+@interface STPAPIClient (PaymentIntentPrivate)
+
+- (void)retrievePaymentIntentWithClientSecret:(NSString *)secret
```
Comment (yuki-stripe) [inline Stripe/STPAPIClient+Private.h]: can we document these methods?

File: Stripe/PublicHeaders/STPPaymentMethodSEPADebit.h [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+ */
+@property (nonatomic, nullable, readonly) NSString *country;
+
+/**
+ The account's fingerprint.
+ */
+@property (nonatomic, nullable, readonly) NSString *fingerprint;
+
+/**
+ The reference of the mandate accepted by your customer.
```
Comment (yuki-stripe) [inline Stripe/PublicHeaders/STPPaymentMethodSEPADebit.h]: Not clear what a mandate is here, should we link to a Stripe object?

**Pull request merged**